test/expected/ConvertToBin/Main.vm

```diff
@@ -69,3 +69,43 @@ goto WHILE_COND0
 label WHILE_END0
 push constant 0
 return
+function Main.nextMask 0
+push argument 0
+push constant 0
+eq
+if-goto IF_TRUE2
```

Comment:
Right, that would explain why the book's implementation of the compiler does reset the label counter/index for each function -- because they assume the VM translator would take care of the scope.